src/app/settings/action-triggers/components/action-triggers-list/action-triggers-list.component.ts

```diff
@@ -0,0 +1,240 @@
+        this.actionTriggersWithAction = this.actionTriggersWithAction.sort((a, b) => b.actionTrigger.data?.keys.toString().localeCompare(a.actionTrigger.data?.keys.toString()))
+        break
+      default:
+        this.actionTriggersWithAction = this.actionTriggersWithAction.sort((a, b) => a.actionTrigger.actionId.localeCompare(b.actionTrigger.actionId))
+        break
+    }
+  }
+
+  public actionWithSelected(userAction: string): void {
+    switch (userAction) {
+      case ActionsWithSelected.DELETE:
+        return this.dialogService.createConfirmDialog($localize`action-triggers.delete-selected.label`, $localize`action-triggers.delete-selected.confirmation`, 'Delete', () =>
+          this.deleteSelectedActionTriggers()
+        )
+      case ActionsWithSelected.EXPORT:
+        return this.exportSelectedTriggers()
+      case ActionsWithSelected.TOGGLE_SELECT:
+        return this.toggleSelectUnselectAll()
+      default:
+        return
+    }
+  }
+
+  private updateSelectedActionTriggersOptions(): void {
+    this.selectedActionTriggerOptions = this.selectedActionTriggerOptions.map(this.updateSelectedActionTriggersOption.bind(this))
+  }
+
+  private updateSelectedActionTriggersOption(option: SofieDropdownOption): SofieDropdownOption {
+    switch (option.key) {
+      case ActionsWithSelected.TOGGLE_SELECT:
+        return {
+          ...option,
+          label: this.areAllActionTriggersSelected() ? $localize`global.unselect-all.label` : $localize`global.select-all.label`,
+        }
+      case ActionsWithSelected.EXPORT:
+      case ActionsWithSelected.DELETE:
+        return {
+          ...option,
+          isDisabled: this.isNoActionTriggerSelected(),
+        }
+      default:
+        return option
+    }
+  }
+
+  private areAllActionTriggersSelected(): boolean {
+    return this.actionTriggersWithAction.length === this.selectedActionTriggerIds.size
+  }
+
+  private isNoActionTriggerSelected(): boolean {
+    return this.selectedActionTriggerIds.size === 0
+  }
+
+  public isActionTriggerSelected(id: string): boolean {
+    return this.selectedActionTriggerIds.has(id)
+  }
+
+  private exportSelectedTriggers(): void {
+    const selectedTriggers: ActionTrigger<KeyboardTriggerData>[] = []
+    this.selectedActionTriggerIds.forEach(actionTriggerId => {
+      const actionTrigger: ActionTrigger<KeyboardTriggerData> | undefined = this.actionTriggersWithAction.find(actionTriggerItem => actionTriggerItem.actionTrigger.id === actionTriggerId)
+        ?.actionTrigger
+      if (!actionTrigger) {
+        return
+      }
+      selectedTriggers.push({
+        actionId: actionTrigger.actionId,
+        id: actionTrigger.id,
+        data: {
+          keys: actionTrigger.data.keys,
+          actionArguments: actionTrigger.data.actionArguments,
+          label: actionTrigger.data.label,
+          triggerOn: actionTrigger.data.triggerOn,
+          mappedToKeys: actionTrigger.data.mappedToKeys,
+        },
+      })
+    })
```

Comment:
Like with `cloneActionTrigger`, spreading the objects should be enough for cloning them.
But I don't think in this case that the cloning is necessary. I think you can get away with a filter and a map like so:
```typescript
  private exportSelectedActionTriggers(): void {
    const selectedActionTriggers: ActionTrigger<KeyboardTriggerData>[] = this.actionTriggersWithAction
      .filter(actionTriggerWithAction => this.selectedActionTriggerIds.has(actionTriggerWithAction.actionTrigger.id))
      .map(actionTriggerWithAction => actionTriggerWithAction.actionTrigger)
    this.fileDownloadService.downloadText(JSON.stringify(selectedActionTriggers), 'selected-actions-triggers.json')
    this.unselectAllActionsTriggers()
    this.updateSelectedActionTriggersOptions()
  }
```
I have renamed the method to `exportSelectedActionTriggers` instead of `exportSelectedTriggers`.

It can also be considered to move the filter and map out into a method called something like `getSelectedActionTriggers`, to make the intent of this method clearer.

Reply:
Done